packages/happy-dom/src/nodes/html-script-element/HTMLScriptElement.ts

```diff
@@ -189,12 +189,18 @@ export default class HTMLScriptElement extends HTMLElement implements IHTMLScrip
 						type === 'application/x-javascript' ||
 						type.startsWith('text/javascript'))
 				) {
-					if (this.ownerDocument.defaultView.happyDOM.settings.disableErrorCapturing) {
-						this.ownerDocument.defaultView.eval(textContent);
-					} else {
-						WindowErrorUtility.captureError(this.ownerDocument.defaultView, () =>
-							this.ownerDocument.defaultView.eval(textContent)
-						);
+					try {
```

Comment:
We should not do a try and catch here. Thats why we have the `WindowErrorUtility.captureError()` method that only should be used if `this.ownerDocument.defaultView.happyDOM.settings.disableErrorCapturing` is set to false.

If error capturing is disabled, we want the error to be thrown and catched by the framework using Happy DOM.

Reply:
We should also add the `this.ownerDocument['_currentScript'] = *` logic to `HTMLScriptElementUtility` when it is evaluating the code.

Reply:
Thanks for the feedback!

> We should not do a try and catch here. Thats why we have the `WindowErrorUtility.captureError()` method that only should be used if `this.ownerDocument.defaultView.happyDOM.settings.disableErrorCapturing` is set to false.
> 
> If error capturing is disabled, we want the error to be thrown and catched by the framework using Happy DOM.

This `try` does not actually catch any errors because there isn't a `catch`-block, and it just has a `finally`-block so we can set `this.ownerDocument['_currentScript']` to `null`. So any errors should passed through as before.

> We should also add the this.ownerDocument['_currentScript'] = * logic to HTMLScriptElementUtility when it is evaluating the code.

I will fix that! 👍

Reply:
The try block slows down execution a lot as it will still create a new context for each try to capture errors.

If we encounter and error when `this.ownerDocument.defaultView.happyDOM.settings.disableErrorCapturing` is set to true, I think it is expected that the execution stops anyway.

You can see a performance test of try/finally here:
[Link to test](https://perf.link/#eyJpZCI6Ims5aGVmd2Y3amt5IiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6IiIsInRlc3RzIjpbeyJuYW1lIjoidHJ5L2ZpbmFsbHkiLCJjb2RlIjoiICAgIHRyeSB7XG4gICAgICAgIEpTT04ucGFyc2UoJ3tcIm5hbWVcIjpcIkpvaG5cIiwgXCJhZ2VcIjozMSwgXCJjaXR5XCI6XCJOZXcgWW9ya1wifScpO1xuICAgIH0gZmluYWxseSB7XG4gICAgICAgIGNvbnNvbGUubG9nKCdmaW5hbGx5Jyk7XG4gICAgfSIsInJ1bnMiOlsyMDUwMDAsMTQwMDAwLDcwMDAsMTAwMCw4MzAwMCw1MTAwMCwyMzcwMDAsMjY2MDAwLDg3MDAwLDI4NzAwMCwzMDEwMDAsMTAwMCwzMjQwMDAsOTMwMDAsMTAwMDAsODAwMCwyNTMwMDAsMTMwMDAwLDMzNTAwMCwyMjYwMDAsNjEwMDAsMTAwMDAsMjY4MDAwLDM0NDAwMCwyNDEwMDAsMjU1MDAwLDI5MDAwMCwzMzkwMDAsMTA3MDAwLDEyNjAwMCw5MTAwMCw4NDAwMCwxNDYwMDAsMTY1MDAwLDkxMDAwLDgzMDAwLDE0NjAwMCwxMTAwMCwxNDkwMDAsNjIwMDAsMjE2MDAwLDIzMjAwMCwxMjAwMCwxMjgwMDAsMjA0MDAwLDQ2MDAwLDIwNjAwMCwyMzcwMDAsMTE3MDAwLDI2MDAwLDI2NTAwMCwxMjMwMDAsOTgwMDAsMjQxMDAwLDU3MDAwLDEwNzAwMCwyMjAwMCwxMDYwMDAsMTEwMDAsMTUxMDAwLDI4ODAwMCwxMDAwLDI2NjAwMCw3OTAwMCwxMDAwMDAsMjYwMDAsMTEzMDAwLDIyMTAwMCw5MjAwMCwxOTYwMDAsMTAxMDAwLDE0MDAwLDIxMDAwMCwxODUwMDAsMTcwMDAsMzAxMDAwLDEyNzAwMCwxNzgwMDAsMjgwMDAsMzAwMDAsMzE1MDAwLDIwMzAwMCwyMTIwMDAsMTI3MDAwLDEwMDAwMCwyMzcwMDAsMjE0MDAwLDE0MTAwMCw0MzAwMCwxNTgwMDAsNzAwMDAsMjg3MDAwLDIwNTAwMCwxNzAwMDAsMjg2MDAwLDE3MzAwMCw2NTAwMCwyNDAwMCwxMDAwMDAsMTAwMF0sIm9wcyI6MTQ0MjMwfSx7Im5hbWUiOiJ3aXRob3V0IHRyeS9maW5hbGx5IiwiY29kZSI6IkpTT04ucGFyc2UoJ3tcIm5hbWVcIjpcIkpvaG5cIiwgXCJhZ2VcIjozMSwgXCJjaXR5XCI6XCJOZXcgWW9ya1wifScpOyIsInJ1bnMiOlsxMzk1MDAwLDk4MDAwMCwxMDkwMDAsMTc5MzAwMCw0OTYwMDAsNDI5MDAwLDg5MjAwMCwxODY0MDAwLDY5MDAwMCwyMDUyMDAwLDE0NjYwMDAsMTM2NTAwMCwxNDY1MDAwLDQ0NjAwMCw3MDAwLDE0NjUwMDAsMTA3NDAwMCwxMDkwMDAwLDIxOTEwMDAsMTY1ODAwMCwxNzUyMDAwLDE1MDAwMCwxNDkyMDAwLDE1MjQwMDAsMTU1MjAwMCwxNDY1MDAwLDEzOTcwMDAsMTk0MzAwMCw5MDAwMDAsNDc4MDAwLDEzOTcwMDAsNzQ5MDAwLDExNzAwMDAsMTE1NDAwMCw0MzcwMDAsMTI2OTAwMCwxNTc2MDAwLDE4NDAwMDAsMTkyNDAwMCw1MjkwMDAsMTgwODAwMCwyMTQzMDAwLDE0MTAwMCwyMDAwMCwxOTcwMDAwLDE3NjAwMCw1NjAwMCwyMTg5MDAwLDg5NDAwMCwyMjgwMDAsMTQ2NTAwMCwxMTE5MDAwLDIwNDAwMCwxNjk4MDAwLDEwMzEwMDAsMTYxMDAwLDIxODkwMDAsMjQwODAwMCwxNDIwMDAsMTM0NzAwMCwxNTE1MDAwLDE2NDYwMDAsMTI1OTAwMCw2MDgwMDAsMzU2MDAwLDcxMzAwMCw5NTAwMCwxNDY1MDAwLDIwOTAwMCwxNDY1MDAwLDc1MzAwMCwxMzIxMDAwLDQyNzAwMCwzMzcwMDAsMjMyMDAwLDIxNzIwMDAsMTgwNDAwMCwxMTEzMDAwLDkyNTAwMCwyMTYzMDAwLDE4ODUwMDAsODE5MDAwLDIzMDkwMDAsNzAwMDAsMjAwNTAwMCw5NTAwMCwxNjEwMDAwLDEyNDUwMDAsNDQzMDAwLDE0NjUwMDAsMTQ2NTAwMCwyMTg5MDAwLDc5MzAwMCwyMjcxMDAwLDUyNzAwMCwxMDIwMDAwLDYzNTAwMCwyNDAwMCw0NzAwMDAsMTM2MDAwXSwib3BzIjoxMTEwNjgwfV0sInVwZGF0ZWQiOiIyMDIzLTA5LTI5VDEwOjM2OjI4LjIyNloifQ%3D%3D)

Reply:
Ahh I see, thanks for the link and info. Didn't know that. I will change it then 🙂